### PR TITLE
fix: [CDS-37814]: fixed the exec strategy for gitops

### PR DESCRIPTION
**Pull request by sahithikolichala**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 1]
```diff
@@ -74,6 +74,13 @@ const TabsOrder = [
 
 const iconNames = { tick: 'tick' as IconName }
 
+const getStrategyType = (selectedStage: any, selectedDeploymentType: any): any => {
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:77]: We have proper types for These. Please use them

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 2]
```diff
@@ -175,17 +182,19 @@ export default function DeployStageSetupShell(): JSX.Element {
     }
   }, [selectedTabId])
 
-  const selectedDeploymentType = getSelectedDeploymentType(
-    selectedStage,
-    getStageFromPipeline,
-    !!selectedStage?.stage?.spec?.serviceConfig?.useFromStage?.stage
-  )
+  const selectedDeploymentType = isNewServiceEnvEntity(NG_SVC_ENV_REDESIGN, selectedStage as any)
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:185]: Use proper type for selectedStage

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 3]
```diff
@@ -175,17 +182,19 @@ export default function DeployStageSetupShell(): JSX.Element {
...
+    ? (selectedStage?.stage?.spec as any)?.deploymentType
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:186]: Here also

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 4]
```diff
@@ -175,17 +182,19 @@ export default function DeployStageSetupShell(): JSX.Element {
...
+        !!selectedStage?.stage?.spec?.serviceConfig?.useFromStage?.stage
+      )
   const { data: stageYamlSnippet, loading, refetch } = useGetFailureStrategiesYaml({ lazy: true })
 
   const { data: yamlSnippet, refetch: refetchYamlSnippet } = useGetExecutionStrategyYaml({
     queryParams: {
-      serviceDefinitionType: selectedDeploymentType,
-      strategyType: selectedDeploymentType === ServiceDeploymentType.ServerlessAwsLambda ? 'Basic' : 'Rolling'
+      serviceDefinitionType: selectedDeploymentType || (selectedStage?.stage?.spec as any)?.deploymentType,
+      strategyType: getStrategyType(selectedStage, selectedDeploymentType)
```
Comment (bhavya-sinha) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:197]: Here. We have exposed a new type `DeployStageConfig` in `src/modules/70-pipeline/utils/DeployStageInterface.ts` You can use that

**Review by bhavya-sinha — CHANGES_REQUESTED** [review]

**Review by chetanvekariya-harness — DISMISSED** [review]
Changes seem fine regarding execution strategies panel not to be opened when git ops is enabled.

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 5]
```diff
@@ -175,17 +182,19 @@ export default function DeployStageSetupShell(): JSX.Element {
...
+        getStageFromPipeline,
+        !!selectedStage?.stage?.spec?.serviceConfig?.useFromStage?.stage
+      )
   const { data: stageYamlSnippet, loading, refetch } = useGetFailureStrategiesYaml({ lazy: true })
 
   const { data: yamlSnippet, refetch: refetchYamlSnippet } = useGetExecutionStrategyYaml({
     queryParams: {
-      serviceDefinitionType: selectedDeploymentType,
-      strategyType: selectedDeploymentType === ServiceDeploymentType.ServerlessAwsLambda ? 'Basic' : 'Rolling'
+      serviceDefinitionType: selectedDeploymentType || (selectedStage?.stage?.spec as any)?.deploymentType,
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:196]: If selectedDeploymentType could be null / undefined then value of `selectedStage?.stage?.spec as any)?.deploymentType` must also be undefined.
So, I wanted to ask if this `||` condition is required ?

File: src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx [hunk 6]
```diff
@@ -285,11 +294,13 @@ export default function DeployStageSetupShell(): JSX.Element {
           const stageType = selectedStage?.stage?.type
           const openExecutionStrategy = stageType ? stagesMap[stageType].openExecutionStrategy : true
           const isServerlessDeploymentTypeSelected = isServerlessDeploymentType(selectedDeploymentType)
+          const gitOpsEnabled = (selectedStage?.stage?.spec as any)?.gitOpsEnabled
           // Show executiomn strategies when openExecutionStrategy is true and deployment type is not serverless
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineStudio/DeployStageSetupShell/DeployStageSetupShell.tsx:298]: This comment becomes outdated now
`// Show executiomn strategies when openExecutionStrategy is true and deployment type is not serverless`

You may update comment as you have added one more check now. Will be great if we can take this condition into some function where we can keep adding more conditions as required.

**Review by bhavya-sinha — APPROVED** [review]

**Review by vkbansal — APPROVED** [review]

**Pull request merged**